### PR TITLE
analyzer: Add recognition of missing HTTP values

### PR DESCRIPTION
**Pull request by jlucovsky**
This changeset eliminates warnings based on (private) a specific set of rules. Those rule highlighted missing HTTP values that were not recognized by the engine analyzer.

Link to [2926](https://redmine.openinfosecfoundation.org/issues/2926) ticket:

Describe changes:
- Recognize additional HTTP values: server body, raw host, user agent, header names

### CODE REVIEW

File: src/detect-http-server-body.c [hunk 1]
```diff
@@ -91,7 +91,7 @@ void DetectHttpServerBodyRegister(void)
     sigmatch_table[DETECT_HTTP_RESPONSE_BODY].flags |= SIGMATCH_NOOPT;
     sigmatch_table[DETECT_HTTP_RESPONSE_BODY].flags |= SIGMATCH_INFO_STICKY_BUFFER;
 
-    g_file_data_buffer_id = DetectBufferTypeRegister("file_data");
+    g_http_server_body_buffer_id = DetectBufferTypeRegister("http_server_body");
```
Comment (victorjulien) [inline src/detect-http-server-body.c]: this registration as 'file_data' is intentional. The file_data buffer inspect http server body in HTTP, file data in SMTP attachments, etc. The http_server_body modifier implements file_data for http.

File: doc/userguide/rules/http-keywords.rst [hunk 2]
```diff
@@ -60,8 +60,8 @@ http_header                    Modifier                 Both
 http_raw_header                Modifier                 Both
 http_cookie                    Modifier                 Both
 http_server_body               Modifier                 Response
-http.server                    Modifier                 Response
-http.location                  Modifier                 Response
+http.server                    Sticky Buffer            Response
```
Comment (victorjulien) [inline doc/userguide/rules/http-keywords.rst]: this is not related to the analyzer, so should go into its own commit

File: src/detect-http-server-body.c (hunk 1 above)
Comment (jlucovsky) [reply src/detect-http-server-body.c]: Will revert these changes.

File: doc/userguide/rules/http-keywords.rst (hunk 2 above)
Comment (jlucovsky) [reply doc/userguide/rules/http-keywords.rst]: Will do.

**Comment by jlucovsky** [comment]
Continued in #3885
